File: flowcore/src/main/java/com/grosner/dbflow/sql/SQLiteType.java
package com.grosner.dbflow.sql;

import java.util.HashMap;

/**
 * Author: andrewgrosner
 * Description: Holds the mapping between SQL data types and java classes.
 */
public enum SQLiteType {

    INTEGER,
    REAL,
    TEXT,
    BLOB;

    private static final HashMap<String, SQLiteType> sTypeMap = new HashMap<String, SQLiteType>() {
        {
            put(byte.class.getName(), SQLiteType.INTEGER);
            put(short.class.getName(), SQLiteType.INTEGER);
            put(int.class.getName(), SQLiteType.INTEGER);
            put(long.class.getName(), SQLiteType.INTEGER);
            put(float.class.getName(), SQLiteType.REAL);
            put(double.class.getName(), SQLiteType.REAL);
            put(boolean.class.getName(), SQLiteType.INTEGER);
            put(char.class.getName(), SQLiteType.TEXT);
            put(byte[].class.getName(), SQLiteType.BLOB);
            put(Byte.class.getName(), SQLiteType.INTEGER);
            put(Short.class.getName(), SQLiteType.INTEGER);
            put(Integer.class.getName(), SQLiteType.INTEGER);
            put(Long.class.getName(), SQLiteType.INTEGER);
            put(Float.class.getName(), SQLiteType.REAL);
            put(Double.class.getName(), SQLiteType.REAL);
            put(Boolean.class.getName(), SQLiteType.INTEGER);
            put(Character.class.getName(), SQLiteType.TEXT);
            put(String.class.getName(), SQLiteType.TEXT);
            put(Byte[].class.getName(), SQLiteType.BLOB);
        }
    };

    /**
     * Returns the {@link SQLiteType} for this class
     *
     * @param className The fully qualified class name
     * @return
     */
    public static SQLiteType get(String className) {
        return sTypeMap.get(className);
    }

    public static boolean containsClass(String className) {
        return sTypeMap.containsKey(className);
    }
}

File: dbflow-core/src/main/java/com/raizlabs/dbflow5/sql/SQLiteType.kt
package com.raizlabs.dbflow5.sql

import com.raizlabs.dbflow5.data.Blob

actual val platformTypeMap: Map<String, SQLiteType> = hashMapOf(
    ByteArray::class.java.name to SQLiteType.BLOB,
    Byte::class.java.name to SQLiteType.INTEGER,
    Short::class.java.name to SQLiteType.INTEGER,
    Int::class.java.name to SQLiteType.INTEGER,
    Long::class.java.name to SQLiteType.INTEGER,
    Float::class.java.name to SQLiteType.REAL,
    Double::class.java.name to SQLiteType.REAL,
    Boolean::class.java.name to SQLiteType.INTEGER,
    Char::class.java.name to SQLiteType.TEXT,
    CharSequence::class.java.name to SQLiteType.TEXT,
    String::class.java.name to SQLiteType.TEXT,
    Array<Byte>::class.java.name to SQLiteType.BLOB,
    Blob::class.java.name to SQLiteType.BLOB,
    Byte::class.javaPrimitiveType!!.name to SQLiteType.INTEGER,
    Short::class.javaPrimitiveType!!.name to SQLiteType.INTEGER,
    Int::class.javaPrimitiveType!!.name to SQLiteType.INTEGER,
    Long::class.javaPrimitiveType!!.name to SQLiteType.INTEGER,
    Float::class.javaPrimitiveType!!.name to SQLiteType.REAL,
    Double::class.javaPrimitiveType!!.name to SQLiteType.REAL,
    Boolean::class.javaPrimitiveType!!.name to SQLiteType.INTEGER,
    Char::class.javaPrimitiveType!!.name to SQLiteType.TEXT
)
